lib/src/controls/form/password_box.dart

```diff
@@ -0,0 +1,284 @@
+import 'package:fluent_ui/fluent_ui.dart';
+
+enum PasswordRevealMode {
+  /// The password reveal button is visible. The password is not obscured while
+  /// the button is pressed.
+  ///
+  /// If the focus is lost, the button will be hidden on next time the focus
+  /// is got until the password box is cleared. (This is a security concern).
+  ///
+  /// If you want to keep the reveal button visible, see [peekAlways].
+  peek,
+
+  /// The password reveal button is visible. The button is not obscured while
+  /// the button is pressed.
+  ///
+  /// The reveal button will always be visible if the password box has the focus
+  /// and it's not empty. Unlike the [peek] mode, if the focus is regained,
+  /// the reveal button will be visible.
+  peekAlways,
+
+  /// The password reveal button is not visible. The password is
+  /// always obscured.
+  hidden,
+
+  /// The password reveal button is not visible. The password is not obscured.
+  visible,
+}
+
+/// A fluent design input form for password.
+///
+/// A PasswordBox lets the user enter a password. There is multiple mode to
+/// indicate if the password must be hidden, visible or if a reveal button is
+/// shown.
+///
+/// See also:
+///
+///  * https://learn.microsoft.com/en-us/windows/apps/design/controls/password-box
+class PasswordBox extends StatefulWidget {
+  /// Controls the text being edited.
+  ///
+  /// If null, this widget will create its own [TextEditingController].
+  final TextEditingController? controller;
+
+  /// Disables the text field when false.
+  ///
+  /// Text fields in disabled states have a light grey background and don't
+  /// respond to touch events including the [prefix], [suffix].
```

Comment:
We don't have prefix nor suffix

```suggestion
  /// respond to touch events.
```